fri/src/two_adic_pcs.rs

```diff
@@ -106,26 +108,26 @@ impl<F: TwoAdicField, InputProof, InputError: Debug> FriGenericConfig<F>
         //     p_o(g^(2i)) = (p(g^i) - p(g^(n/2 + i))) / (2 g^i)
         // so
         //     result(g^(2i)) = p_e(g^(2i)) + beta p_o(g^(2i))
-        //                    = (1/2 + beta/2 g_inv^i) p(g^i)
-        //                    + (1/2 - beta/2 g_inv^i) p(g^(n/2 + i))
+        //
+        // As p_e, p_o will be in the extension field we want to find ways to avoid extension multiplications.
+        // We should only need a single one (namely multiplication by beta).
         let g_inv = F::two_adic_generator(log2_strict_usize(m.height()) + 1).inverse();
-        let one_half = F::ONE.halve();
-        let half_beta = beta * one_half;
 
         // TODO: vectorize this (after we have packed extension fields)
 
-        // beta/2 times successive powers of g_inv
-        let mut powers = g_inv
-            .shifted_powers(half_beta)
+        // As beta is in the extension field, we want to avoid multiplying by it
+        // for as long as possible. Hence we use powers instead of shifted_powers.
```

Comment:
Nit: This is confusing because you _do_ use `.shifted_powers`, just in the base field.
```suggestion
        // for as long as possible. Here we precompute the powers  `g_inv^i / 2` in the base field.
```